dbcrossbarlib/src/tls.rs

```diff
@@ -0,0 +1,30 @@
+//! Support for setting up RusTLS in a consistent fashion.
+
+use rustls::{Certificate, ClientConfig, RootCertStore};
+use rustls_native_certs::load_native_certs;
+
+use crate::common::*;
+
+/// Standard RusTLS `ClientConfig` setup.
+///
+/// We hope to be able to reuse this for multiple different types of TLS
+/// connections.
+pub(crate) fn rustls_client_config(ctx: &Context) -> Result<ClientConfig> {
+    // Set up RusTLS.
+    let mut root_store = RootCertStore::empty();
+    for cert in load_native_certs().context("could not find system cert store")? {
+        root_store
+            .add(&Certificate(cert.0))
+            .context("could not add certificate to cert store")?;
+    }
+    for subject in root_store.subjects() {
+        let bytes = subject.0.iter().map(|n| *n as u8).collect::<Vec<_>>();
```

Comment:
why do we need the `map` with the dereference? is it necessary for the cast to `u8`?

Reply:
I should probably just remove that code—I was using it to print out the certs we loaded. We have to do a weird song and dance here because of x509 nonsense, mostly involving unparsed indentifiers that we don't have good parsing utilities for.

Reply:
Since we will someday need that horrible debugging hack again, I have broken my rule against leaving commented out code, and just commented it out.

We have to cast to `u8` because we're dealing with a mix of UTF-16 Unicode data and binary framing data, and we're just pulling out things that look ASCII-like and printing everything. :-( It was not nice code.

Reply:
I actually don't think it's the worst to leave it flagged behind extra verbose logging, like `dbcrossbar -vvv cp ...`.

Reply:
if I'm asking for 3x verbosity then I shouldn't be surprised when the tool does something like print out every considered SSL cert :slightly_smiling_face:

Reply:
I mean, it was already `trace`, the lowest level of logging. But the code was pretty unforgivable and undocumented.